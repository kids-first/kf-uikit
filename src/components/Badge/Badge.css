.Badge {
  @apply w-auto
    m-4
    py-4
    px-20
    rounded-full
    text-center
    text-xs
    text-white
    leading-tight
    whitespace-no-wrap;
}

.Badge--loading {
  @apply bg-lightGrey;
}
