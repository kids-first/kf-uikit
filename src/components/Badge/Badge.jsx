import React from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';

/**
 * A Simple rounded button
 */
const Badge = ({ size, color, className, children }) => {

  const buttonClass = classes(
    'Badge',
    {[`Badge--${size}`]: ['small', 'large'].includes(size)},
    `Badge--${color}`,
    className
  );
  return (
    <span className={buttonClass}>
      {children}
    </span>
  );
}

Badge.propTypes = {
  /** The size of the button */
  size: propTypes.string,
  /** The color of the button. */
  color: propTypes.string,
  /** Any additional classes to be applied to the button */
  className: propTypes.string,
  /** Children elements. */
  children: propTypes.string,
};

Badge.defaultProps = {
  size: 'default',
  color: 'primary',
  className: null,
  children: null,
};

export default Badge;
