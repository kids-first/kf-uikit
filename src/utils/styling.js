import {
  space,
  width,
  fontSize,
  color,
  lineHeight,
  fontWeight,
  borders,
  borderRadius,
  justifySelf,
  alignSelf,
  maxWidth,
  complexStyle,
  hover,
  textAlign,
} from 'styled-system';
import styled from '../kfEmotion';

export const applyProp = (name, value) => (value ? `${name}: ${value};` : '');
export const overflow = ({ ov }) => applyProp('overflow', ov);
export const overflowY = ({ ov }) => applyProp('overflow-y', ov);
const baseStyles = complexStyle({
  prop: 'baseStyle',
  key: 'baseStyles',
});

export const applyDefaultStyles = Component => styled(Component)`
  ${baseStyles}
  ${space}
  ${width}
  ${fontSize}
  ${fontWeight}
  ${color}
  ${maxWidth}
  ${justifySelf}
  ${alignSelf}
  ${borders}
  ${borderRadius}
  ${lineHeight}
  ${overflow}
  ${overflowY}
  ${hover}
  ${textAlign}
`;
