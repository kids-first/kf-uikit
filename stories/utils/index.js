import { withInfo as withAddonInfo } from '@storybook/addon-info';
import { find } from 'lodash';

const mdListFromArr = arr => arr.map(
  thing => `
    - ${thing}
  `,
)
  .join(',')
  .replace(/,/g, ' ');

export const componentShould = ({
  preText = '',
  doThis = [],
  notThis = [],
  guidelines = '',
  postText = '',
}) => `
  ${preText}

  #### Best Practices
  This component **should**: 
  ${mdListFromArr(doThis)}
  
  This component **should NOT**: 
  ${mdListFromArr(notThis)}

  #### Content Guidelines 
  ${guidelines}

  ${postText}
`;

export const withInfo = (options, component) => {
  const SRC = find(global.STORYBOOK_REACT_CLASSES, {
    name: component().type.name,
  });

  return withAddonInfo(
    Object.assign(
      {},
      {
        text: SRC ? SRC.docgenInfo.description : '',
        inline: true,
        header: false,
      },
      options,
    ),
  )(component);
};
