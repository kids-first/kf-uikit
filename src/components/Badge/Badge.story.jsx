/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { select, text } from '@storybook/addon-knobs';
import Badge from './Badge';

const stories = storiesOf('Badges', module);

stories.add('Simple Badge', () => (
  <Badge
    size={select('Size', ['small', 'default', 'large'], 'default')}
    color={select('Color', ['primary', 'secondary', 'tertiary', 'error', 'grey'], 'grey')}
  >
    {text('Text', '10,567')}
  </Badge>
  ),
  {
		info: {
      text: `
        Used to show a numeric stat or icon.
        `
    },
  }
)
