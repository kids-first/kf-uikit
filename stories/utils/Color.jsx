import React from 'react';
import propTypes from 'prop-types';
import { space, color, fontSize } from 'styled-system';
import chroma from 'chroma-js';
import styled from '../../src/kfEmotion';
import { H3, P } from '../../src/components/Typography';
import { Box } from '../../src/components/Layout';
import { wcagGrade } from '../../src/utils/colors';

/**
 * A basic square swatch to display a color
 */
/* eslint-disable no-shadow */
const Swatch = styled('div')`
  width: 100px;
  height: 100px;
  display: block;
  border-radius: 4px;
  margin: auto;
  background-color: ${({ color }) => color};
  ${space} ${color};
`;

Swatch.propTypes = {
  ...space.propTypes,
  ...color.propTypes,
};

/**
 * A basic square swatch to display a gradient
 */
const GradientSwatch = styled(Swatch)`
  width: 256px;
  background-color: none;
  background-image: ${({ gradient }) => gradient};
`;
/**
 * A simple badge
 */
const Badge = styled('span')`
  border-radius: 4px;
  border-width: 1px;
  border-color: '#333';
  padding: 3px;
  margin: 5px;
  display: block;
  ${color} ${fontSize};
`;
Badge.propTypes = {
  ...space.propTypes,
  ...color.propTypes,
};

/**
 * Displays a color swatch with its hex and rgba values along with a type
 * specimen on a badge of that color
 */

const ColorSpecimen = (
  { color, name }, // eslint-disable-line no-shadow
) => (
  <Box m={24} w={256} display="inline-block">
    <Swatch color={chroma(color).css()} />
    <H3 mb={1}>{name}</H3>
    <P mt={0} mb={0} small>
      hex: {chroma(color).hex()}
    </P>
    <P mt={0} small>
      rgb: {chroma(color).css()}
    </P>
    <Badge color="black" bg={chroma(color).css()}>
      {wcagGrade('black', color)}
    </Badge>
    <Badge color="white" bg={chroma(color).css()}>
      {wcagGrade('white', color)}
    </Badge>
  </Box>
);

ColorSpecimen.propTypes = {
  name: propTypes.string,
  color: propTypes.string,
};

ColorSpecimen.defaultProps = {
  name: 'Color',
  color: '#999',
};

/**
 * Displays a gradient swatch with title and value
 */
export const GradientSpecimen = ({ gradient, name }) => (
  <Box m={24} w={256} display="inline-block">
    <GradientSwatch gradient={gradient} />
    <H3 mb={1}>{name}</H3>
    <P mt={0} mb={0} small>
      hex: {gradient}
    </P>
  </Box>
);

GradientSpecimen.propTypes = {
  name: propTypes.string,
  gradient: propTypes.string,
};

GradientSpecimen.defaultProps = {
  name: 'Gradient',
  gradient: 'linear-gradient(to right, #fff, #000)',
};

export default ColorSpecimen;
