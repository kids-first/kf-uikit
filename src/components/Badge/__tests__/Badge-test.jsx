import React from 'react';
import renderer from 'react-test-renderer';
import { generateVariantsFor } from '../../../utils/propsVariants';
import { toPairs } from 'lodash';
import Badge from '../Badge';
import * as theme from '../../../../tailwind';

const variants = generateVariantsFor({
  color: Object.keys(theme.colors),
  loading: [true, false],
  text: ['valid', 'invalid'],
});

variants.map(props => {
  it(`Badge ${toPairs(props).map(prop => `${prop[0]}:${prop[1]}, `)} renders correctly`, () => {
    const tree = renderer.create(<Badge {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
