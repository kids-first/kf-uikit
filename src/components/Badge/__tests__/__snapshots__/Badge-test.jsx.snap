// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default error badge renders correctly 1`] = `
<span
  className="Badge Badge--error"
>
  Hello
</span>
`;

exports[`default error badge renders correctly 2`] = `
<span
  className="Badge Badge--error"
>
  Hello
</span>
`;

exports[`default grey badge renders correctly 1`] = `
<span
  className="Badge Badge--grey"
>
  Hello
</span>
`;

exports[`default grey badge renders correctly 2`] = `
<span
  className="Badge Badge--grey"
>
  Hello
</span>
`;

exports[`default primary badge renders correctly 1`] = `
<span
  className="Badge Badge--primary"
>
  Hello
</span>
`;

exports[`default primary badge renders correctly 2`] = `
<span
  className="Badge Badge--primary"
>
  Hello
</span>
`;

exports[`default secondary badge renders correctly 1`] = `
<span
  className="Badge Badge--secondary"
>
  Hello
</span>
`;

exports[`default secondary badge renders correctly 2`] = `
<span
  className="Badge Badge--secondary"
>
  Hello
</span>
`;

exports[`default tertiary badge renders correctly 1`] = `
<span
  className="Badge Badge--tertiary"
>
  Hello
</span>
`;

exports[`default tertiary badge renders correctly 2`] = `
<span
  className="Badge Badge--tertiary"
>
  Hello
</span>
`;

exports[`large error badge renders correctly 1`] = `
<span
  className="Badge Badge--large Badge--error"
>
  Hello
</span>
`;

exports[`large error badge renders correctly 2`] = `
<span
  className="Badge Badge--large Badge--error"
>
  Hello
</span>
`;

exports[`large grey badge renders correctly 1`] = `
<span
  className="Badge Badge--large Badge--grey"
>
  Hello
</span>
`;

exports[`large grey badge renders correctly 2`] = `
<span
  className="Badge Badge--large Badge--grey"
>
  Hello
</span>
`;

exports[`large primary badge renders correctly 1`] = `
<span
  className="Badge Badge--large Badge--primary"
>
  Hello
</span>
`;

exports[`large primary badge renders correctly 2`] = `
<span
  className="Badge Badge--large Badge--primary"
>
  Hello
</span>
`;

exports[`large secondary badge renders correctly 1`] = `
<span
  className="Badge Badge--large Badge--secondary"
>
  Hello
</span>
`;

exports[`large secondary badge renders correctly 2`] = `
<span
  className="Badge Badge--large Badge--secondary"
>
  Hello
</span>
`;

exports[`large tertiary badge renders correctly 1`] = `
<span
  className="Badge Badge--large Badge--tertiary"
>
  Hello
</span>
`;

exports[`large tertiary badge renders correctly 2`] = `
<span
  className="Badge Badge--large Badge--tertiary"
>
  Hello
</span>
`;

exports[`small error badge renders correctly 1`] = `
<span
  className="Badge Badge--small Badge--error"
>
  Hello
</span>
`;

exports[`small error badge renders correctly 2`] = `
<span
  className="Badge Badge--small Badge--error"
>
  Hello
</span>
`;

exports[`small grey badge renders correctly 1`] = `
<span
  className="Badge Badge--small Badge--grey"
>
  Hello
</span>
`;

exports[`small grey badge renders correctly 2`] = `
<span
  className="Badge Badge--small Badge--grey"
>
  Hello
</span>
`;

exports[`small primary badge renders correctly 1`] = `
<span
  className="Badge Badge--small Badge--primary"
>
  Hello
</span>
`;

exports[`small primary badge renders correctly 2`] = `
<span
  className="Badge Badge--small Badge--primary"
>
  Hello
</span>
`;

exports[`small secondary badge renders correctly 1`] = `
<span
  className="Badge Badge--small Badge--secondary"
>
  Hello
</span>
`;

exports[`small secondary badge renders correctly 2`] = `
<span
  className="Badge Badge--small Badge--secondary"
>
  Hello
</span>
`;

exports[`small tertiary badge renders correctly 1`] = `
<span
  className="Badge Badge--small Badge--tertiary"
>
  Hello
</span>
`;

exports[`small tertiary badge renders correctly 2`] = `
<span
  className="Badge Badge--small Badge--tertiary"
>
  Hello
</span>
`;
